File: Napakalaki Ruby/lib/cultist_player.rb
# encoding: UTF-8

# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative 'treasure'
require_relative 'monster'
require 'player'

module Napakalaki
  
  class CultistPlayer < Player
    
    attr_reader :totalCultistPlayers
    
    @@totalCultistPlayers = 0
    
    def initialize(player,cultist)
      
      newCopia(player)
      @cultistCard = cultist
      @@totalCultistPlayers += 1
      
    end
    
    def getCombatLevel
      
      return (super + @cultistCard.getSpecialValue)
      
    end
    
    def computeGoldCoinsValue(trs)
      
      return (super(trs) * 2)
      
    end
    
    def getOponentLevel(monster)
      
      return (monster.getSpecialValue)
      
    end
    
    def shouldConvert
      
      return false
      
    end
    
    def self.getTotalCultistPlayers
      
      return @@totalCultistPlayers
      
    end
    
    def to_s

      "#{@name} ,Nivel: #{@level}, Cultist."

    end
    
  end

end

File: Napakalaki Java/src/NapakalakiGame/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

import java.util.ArrayList;

/**
 *
 */
public class CultistPlayer extends Player {
    
    private static int totalCultistPlayers = 0;
    private Cultist myCultistCard;
    
    public CultistPlayer(Player p, Cultist c){
        
        super(p);
        totalCultistPlayers++;
        myCultistCard = c;
    }
    
    public static int getTotalCultistPlayers(){
        
        return totalCultistPlayers;
    }
    
    @Override
    protected int getOpponentLevel(Monster m){
        
        return m.getSpecialValue();
    }
    
    @Override
    protected int getCombatLevel(){
        
        return (super.getCombatLevel() + myCultistCard.getSpecialValue());
    }
    
    @Override
    protected boolean shouldConvert(){
        
        return false;
    }
    
    @Override
    protected float computeGoldCoinsValue (ArrayList<Treasure> trs){
    
        return (super.computeGoldCoinsValue(trs) *2);
        
    }
    
    @Override
    public Cultist getMyCultistCard () {
        
        return myCultistCard;        
    }
}
